src/OSPSuite.Core/Domain/Formulas/FormulaExtensions.cs

```diff
@@ -37,6 +37,15 @@ public static bool IsExplicit(this IFormula formula)
          return formula.IsAnImplementationOf<ExplicitFormula>();
       }
 
+      /// <summary>
+      ///    Returns true if the formula  contains at least one entry using neighborhood reference  otherwise false
+      /// </summary>
+      /// <param name="formula"></param>
+      public static bool IsReferencingNeighborhood(this IFormula formula)
+      {
+         return formula.ObjectPaths.Any(x => x.Contains(ObjectPathKeywords.NBH));
```

Comment:
why `Contains `and not `Equals `?

Reply:
there are many object paths. (mutliple per formula). We want to get the one that contains NBH
== would be wrong because we are not comparing the same type of objects

Reply:
right. I was mistakenly thinking you are checking the entries of a single path, but in fact you are checking multiple paths.